flowmachine/flowmachine/core/server/query_schemas/displacement.py

```diff
@@ -0,0 +1,63 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marshmallow import Schema, fields, post_load
+from marshmallow.validate import OneOf, Length
+from marshmallow_oneofschema import OneOfSchema
+
+from flowmachine.features import Displacement
+from .base_exposed_query import BaseExposedQuery
+from .custom_fields import SubscriberSubset, Statistic
+from .daily_location import DailyLocationSchema, DailyLocationExposed
+
+
+__all__ = ["DisplacementSchema", "DisplacementExposed"]
+
+
+class InputToDisplacementSchema(OneOfSchema):
+    type_field = "query_kind"
+    type_schemas = {"daily_location": DailyLocationSchema}
```

Comment:
Please add modal location to this as well (because `displacement` doesn't care what _kind_ of reference location it receives as input).

Reply:
(Note to self: at some point we likely need some kind of "GenericLocationSchema" which subsumes these, but that's certainly out of scope for this PR.)